src/utils/validationUtil.ts

```diff
@@ -537,10 +544,27 @@ export class ValidationUtil {
     const { testTypes } = payload;
     return !testTypes || !testTypes.length
       ? true
-      : !(testTypes as models.TestType[]).some(
+      : !testTypes.some(
           (testType) =>
             testType.testResult === enums.TEST_RESULT.ABANDONED &&
             !testType.reasonForAbandoning,
         );
   }
+
+  public static ivaFailedHasRequiredFields(testTypes: TestType[]) {
+    if (this.isIvaTest(testTypes)) {
```

Comment:
Is this first if required? As isn't this validation only called if it is an ivaTest? As per loines 47-49.

```
    if (this.isIvaTest(payload.testTypes)) {
      this.ivaFailedHasRequiredFields(payload.testTypes);
    }
```

Reply:
Nice spot, I'll remove that now